ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -5996,7 +5996,7 @@ private ReduceSinkOperator genCommonGroupByPlanReduceSinkOperator(QB qb, List<St
         for (Map.Entry<ASTNode, ExprNodeDesc> entry : nodeOutputs.entrySet()) {
           ASTNode parameter = entry.getKey();
           ExprNodeDesc expression = entry.getValue();
-          if (!(expression instanceof ExprNodeColumnDesc)) {
+          if (!(expression instanceof ExprNodeColumnDesc) && !ExprNodeConstantDesc.isFoldedFromCol(expression)) {
```

Comment:
Do we really care what kind of constant it is? If it is not folded from a column but it is a plain constant what shall we do?

Reply:
Thanks for the elaborate answer. I don't fully grasp the problem but with your description I got a good intuition about it. 

Checking if a constant is folded from a column is more specific than just being a constant in general thus should have smaller impact on existing queries. Since nothing breaks and it solves the problem I am good to go with it.